### PR TITLE
[RSN-45] - Change parameters architecture, replaced role, status, object type with enum

### PR DESCRIPTION
**Pull request by mkoper02**
## Description
Changed _Status_, _Role_, _Object_Type_ into enum in db, created new enum _Type_ and added it to parameters table

## Related issue and/or ticket
RSN-44

### CODE REVIEW

File: Database/init-constraints.sql [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+common.check_fk_exists(object_id INT, object_type common.object_type)
 RETURNS BOOLEAN AS $$
 BEGIN
-	IF object_type_id = 1 THEN
+	IF object_type = 'Event' THEN
 	RETURN EXISTS (SELECT 1 FROM users.user WHERE "id" = object_id);
-	ELSIF object_type_id = 2 THEN RETURN EXISTS (SELECT 1 FROM events.event WHERE "id" = object_id);
+	ELSIF object_type = 'User' THEN RETURN EXISTS (SELECT 1 FROM events.event WHERE "id" = object_id);
 	ELSE RETURN FALSE;
 	END IF;
```
Comment (wzarek) [inline Database/init-constraints.sql:9]: Should't it be that if 'Event' then get from events.event, if 'User' then from users.user

File: Database/init-constraints.sql [hunk 2]
```diff
@@ -45,25 +45,13 @@ ALTER TABLE events.parameter ADD CONSTRAINT chk_parameter_value CHECK (value ~ '
...
-CHECK (
-    (name IN ('Interested', 'Participating') AND object_type_id = 2) OR
-    (name IN ('Completed', 'In progress', 'Waiting for approval') AND object_type_id = 1)
-);
+-- ALTER TABLE common.status
+-- ADD CONSTRAINT chk_object_type_for_status
+-- CHECK (
+--     (name IN ('Interested', 'Participating') AND object_type = 'User') OR
+--     (name IN ('Completed', 'In progress', 'Waiting for approval') AND object_type = 'Event')
+-- );
```
Comment (wzarek) [inline Database/init-constraints.sql:53]: This one we can do directly on events.event and users.user, add check where it's status will be in X. Just like it was but not for status table but for each status column directly check the values of enum.

File: Database/init-db.sql [hunk 3]
```diff
@@ -59,21 +58,18 @@ CREATE TABLE IF NOT EXISTS users.user (
 CREATE TABLE IF NOT EXISTS events.parameter (
   "id" SERIAL PRIMARY KEY,
   "key" text NOT NULL CONSTRAINT events_patameter_key_maxlength CHECK (LENGTH("key") <= 32),
-  "value" text NOT NULL CONSTRAINT events_patameter_value_maxlength CHECK (LENGTH("value") <= 64)
+  "value" text NOT NULL CONSTRAINT events_patameter_value_maxlength CHECK (LENGTH("value") <= 64),
+  "is_predefined" boolean NOT NULL,
+  "type" common.type NOT NULL,
```
Comment (wzarek) [inline Database/init-db.sql:63]: Didn't we agree on making those only in frontend (for proper validation) and leave the database as it was?

**Review by wzarek — CHANGES_REQUESTED** [review]

**Review by wzarek — APPROVED** [review]
OK

**Comment by wzarek** [comment]
@mkoper02 remember to rebase and review conflicts

File: Database/init-db.sql [hunk 4]
```diff
@@ -59,7 +57,7 @@ CREATE TABLE IF NOT EXISTS users.user (
 CREATE TABLE IF NOT EXISTS events.parameter (
   "id" SERIAL PRIMARY KEY,
   "key" text NOT NULL CONSTRAINT events_patameter_key_maxlength CHECK (LENGTH("key") <= 32),
-  "value" text NOT NULL CONSTRAINT events_patameter_value_maxlength CHECK (LENGTH("value") <= 64),
+  "value" text NOT NULL CONSTRAINT events_patameter_value_maxlength CHECK (LENGTH("value") <= 64)
```
Comment (wzarek) [inline Database/init-db.sql:60]: Add comma here :)

**Review by wzarek — CHANGES_REQUESTED** [review]

**Review by wzarek — APPROVED** [review]

**Review by raczu — APPROVED** [review]
lgtm

**Pull request merged**